tidy3d/components/scene.py

```diff
@@ -0,0 +1,1274 @@
+        surfaces, if it is a volume.
+
+        Parameters
+        -------
+        test_object : :class:`.Box`
+            Object for which intersecting media are to be detected.
+        structures : List[:class:`.AbstractMedium`]
+            List of structures whose media will be tested.
+
+        Returns
+        -------
+        List[:class:`.AbstractMedium`]
+            Set of distinct mediums that intersect with the given planar object.
+        """
+        if test_object.size.count(0.0) == 1:
+            # get all merged structures on the test_object, which is already planar
+            structures_merged = Scene._filter_structures_plane_medium(structures, test_object)
+            mediums = {medium for medium, _ in structures_merged}
+            return mediums
+
+        # if the test object is a volume, test each surface recursively
+        surfaces = test_object.surfaces_with_exclusion(**test_object.dict())
+        mediums = set()
+        for surface in surfaces:
+            _mediums = Scene.intersecting_media(surface, structures)
+            mediums.update(_mediums)
+        return mediums
+
+    @staticmethod
+    def intersecting_structures(
+        test_object: Box, structures: Tuple[Structure, ...]
+    ) -> Tuple[Structure, ...]:
+        """From a given list of structures, returns a list of :class:`.Structure` that intersect
+        with the ``test_object``, if it is a surface, or its surfaces, if it is a volume.
+
+        Parameters
+        -------
+        test_object : :class:`.Box`
+            Object for which intersecting media are to be detected.
+        structures : List[:class:`.AbstractMedium`]
+            List of structures whose media will be tested.
+
+        Returns
+        -------
+        List[:class:`.Structure`]
+            Set of distinct structures that intersect with the given surface, or with the surfaces
+            of the given volume.
+        """
+        if test_object.size.count(0.0) == 1:
+            # get all merged structures on the test_object, which is already planar
+            normal_axis_index = test_object.size.index(0.0)
+            dim = "xyz"[normal_axis_index]
+            pos = test_object.center[normal_axis_index]
+            xyz_kwargs = {dim: pos}
+
+            structures_merged = []
+            for structure in structures:
+                intersections = structure.geometry.intersections_plane(**xyz_kwargs)
+                if len(intersections) > 0:
+                    structures_merged.append(structure)
+            return structures_merged
+
+        # if the test object is a volume, test each surface recursively
+        surfaces = test_object.surfaces_with_exclusion(**test_object.dict())
+        structures_merged = []
+        for surface in surfaces:
+            structures_merged += Scene.intersecting_structures(surface, structures)
+        return structures_merged
+
+    """ Plotting General """
+
+    @staticmethod
```

Comment:
I think it's ok for now, but eventually we could probably put a lot of these static methods in viz.py or some `components/scene/viz.py` if we wanted to separate them out.